components/builder-scheduler/src/data_store.rs

```diff
@@ -259,15 +291,19 @@ impl DataStore {
         let mut rng = thread_rng();
         let id = rng.gen::<u64>();
 
-        conn.execute("SELECT insert_group_v1($1, $2)",
-                     &[&(id as i64), &project_names])
+        let (project_names, project_idents): (Vec<String>, Vec<String>) =
+            project_tuples.iter().cloned().unzip();
+
+        conn.execute("SELECT insert_group_v1($1, $2, $3)",
```

Comment:
Yeah I may be mis-understanding here, but are Postgres functions not atomic (and therefore do not support sub-transactions within functions)? Ie, if there is an error within the function and a rollback happens, it will roll back everything everything (ie, both table changes).